src/test/java/edu/ucla/library/bucketeer/verticles/ClearCacheIT.java

```diff
@@ -0,0 +1,372 @@
+            LOGGER.error(failure.getMessage());
+            aContext.fail();
+        });
+    }
+
+    /**
+     * Tests clearing Cantaloupe's cache with null imageID
+     *
+     * @param aContext
+     */
+    @Test
+    public void testNoImageID(final TestContext aContext) {
+        final String imageIDNull = null;
+        final Async asyncTask = aContext.async();
+
+        deployNewVerticle(myConfigs).compose(verticleID -> {
+            myVertID = verticleID;
+            return sendMessage(imageIDNull);
+        }).onFailure(failure -> {
+            if (Integer.parseInt(failure.getMessage()) == HTTP.INTERNAL_SERVER_ERROR) {
+                TestUtils.complete(asyncTask);
+            } else {
+                aContext.fail();
+            }
+        }).onSuccess(success -> {
+            aContext.fail();
+        });
+    }
+
+    /**
+     * Tests that verticles fails when given null credentials
+     *
+     * @param aContext A testing context
+     */
+    @Test
+    public void testNullCredential(final TestContext aContext) {
+        final Async asyncTask = aContext.async();
+        final JsonObject nullConfigs = myConfigs;
+
+        nullConfigs.put(IIIIF_USERNAME, (String) null);
+        nullConfigs.put(IIIIF_PASSWORD, (String) null);
+
+        deployNewVerticle(nullConfigs).onFailure(failure -> {
+            TestUtils.complete(asyncTask);
+        }).onSuccess(success -> {
+            aContext.fail();
+        });
+    }
+
+    /**
+     * Tests that verticles fails when given invalid credentials
+     *
+     * @param aContext A testing context
+     */
+    @Test
+    public void testInvalidCredential(final TestContext aContext) {
+        final Async asyncTask = aContext.async();
+        final JsonObject newConfigs = myConfigs;
+
+        newConfigs.put(IIIIF_USERNAME, "username");
+        newConfigs.put(IIIIF_PASSWORD, "password");
+
+        deployNewVerticle(newConfigs).onFailure(failure -> {
+            TestUtils.complete(asyncTask);
+        }).onSuccess(success -> {
+            aContext.fail();
+        });
+    }
+
+    /**
+     * Gets info.json from Cantaloupe of specific image
+     */
+    private Future<Integer> getInfoJsonObject() {
+        final WebClient client = WebClient.create(Vertx.vertx());
+        final Promise<Integer> promise = Promise.promise();
+
+        LOGGER.debug(myIiifURL + IIIF_RESOURCE);
+
+        client.getAbs(myIiifURL + IIIF_RESOURCE).putHeader(CONTENT_TYPE, JSON).send(request -> {
+            if (request.succeeded()) {
+                final HttpResponse<Buffer> response = request.result();
+
+                if (response.statusCode() == HTTP.OK) {
+                    final String body = response.bodyAsString();
+
+                    try {
+                        promise.complete(new JsonObject(body).getInteger("width"));
+                    } catch (final DecodeException details) {
+                        promise.fail(details);
+                    }
```

Comment:
Since [a 200 response to an info.json request is required to be valid JSON](https://iiif.io/api/image/2.1/#image-information-request), this can be simplified:

```suggestion
                    promise.complete(response.bodyAsJsonObject().getInteger("width"));
```

Reply:
What if something is broken in Cantaloupe and it's not responding as it should per the spec?

Reply:
My bad. I was just looking at the snippet and comment and didn't realize this was in a test. Yes, I agree.